### PR TITLE
Platform.status method

### PR DESCRIPTION
**Pull request by nickray**

### CODE REVIEW

File: src/service.rs [hunk 1]
```diff
@@ -479,13 +481,15 @@ impl<P: Platform> ServiceResources<P> {
             },
 
             Request::RequestUserConsent(request) => {
-                assert_eq!(request.level, consent::Level::Normal);
+                // assert_eq!(request.level, consent::Level::Normal);
```
Comment (alistair23) [inline src/service.rs:484]: Why comment this out?

File: src/store.rs [hunk 2]
```diff
@@ -461,6 +462,7 @@ pub fn read<const N: usize>(store: impl Store, location: Location, path: &Path)
 }
 
 /// Writes contents to path in location of store.
+#[inline(never)]
```
Comment (alistair23) [inline src/store.rs:465]: Why not just let the compiler determine this?

Comment (nickray) [reply src/store.rs:465]: The compiler is bad at optimising for low stack use.

File: src/service.rs (hunk 1 above)
Comment (nickray) [reply src/service.rs:484]: There is little point in panicking in production code. If anything, this should be a debug_assert. Also we handle the Strong case too. Or do you mean I should have deleted the line completely? :)

**Comment by nickray** [comment]
It seems I merged this outside of GitHub, hence closing.

File: src/service.rs (hunk 1 above)
Comment (alistair23) [reply src/service.rs:484]: I meant you should delete the line completely. There isn't much use in having commented out code

File: src/store.rs (hunk 2 above)
Comment (alistair23) [reply src/store.rs:465]: We have tooling in Tock for this. You can pass ` -Z emit-stack-sizes` and then analyse the output, like this: https://github.com/tock/tock/blob/master/boards/Makefile.common#L374

Comment (nickray) [reply src/store.rs:465]: Sweet! I'll have to investigate :)

**Pull request closed without merging**